src/compiler/symbolWalker.ts

```diff
@@ -66,43 +73,22 @@ namespace ts {
                     }
                 }
                 if (type.flags & TypeFlags.TypeParameter) {
-                    visitTypeParameter(type as TypeParameter);
+                    visitType(getConstraintFromTypeParameter(type as TypeParameter));

```

Comment:
I do not really like the inlining of these short functions - it removes the ability to grep for the symbol handling that case, and makes the block less uniform (and potentially makes the outer function more polymorphic). TBQH, if `TypeFlags` wasn't a flags enum, I'd just have stored these visitors in a map and dispatched on kind; but you can't so 🤷‍♂️ .